applications/IgaApplication/custom_strategies/custom_strategies/eigensolver_nitsche_stabilization_strategy.hpp

```diff
@@ -389,95 +389,161 @@ class EigensolverNitscheStabilizationStrategy
+                }
             }
-        }
 
-        SparseMatrixType reduced_stiffness_matrix;
-        reduced_stiffness_matrix = ZeroMatrix((number_of_nodes_master+number_of_nodes_slave)*3,(number_of_nodes_master+number_of_nodes_slave)*3);
+            reduced_stiffness_matrix = ZeroMatrix((number_of_nodes_master+number_of_nodes_slave)*3,(number_of_nodes_master+number_of_nodes_slave)*3);
 
-        for (IndexType i = 0; i < (number_of_nodes_master+number_of_nodes_slave)*3; i++) 
+            for (IndexType i = 0; i < (number_of_nodes_master+number_of_nodes_slave)*3; i++) 
+            {
+                for (IndexType j = 0; j <= i; j++) 
+                {
+                    reduced_stiffness_matrix(i,j) = rStiffnessMatrix(rResult(i),rResult(j));
+                    if (i != j)
+                        reduced_stiffness_matrix(j,i) = rStiffnessMatrix(rResult(i),rResult(j));
+                }
+            }
+        }
+        else // Support Nitsche condition
         {
-            for (IndexType j = 0; j <= i; j++) 
+            // 1. find the DOFs on the current interface boundary
+            Model reduced_model;               
+            ModelPart& reduced_model_part = reduced_model.CreateModelPart("new_model"); 
+
+            for (auto& r_cond : rModelPart.Conditions()) {
+                
+                auto& r_geom = r_cond.GetGeometry();
+                auto& r_N = r_geom.ShapeFunctionsValues();
+
+                for (IndexType i = 0; i<r_N.size2();++i)
+                {
+                    if(r_N(0,i) > 1e-6)
+                    {
+                        reduced_model_part.AddNode(r_geom.pGetPoint(i));
+                    }
+                }
+            }
+            const std::size_t number_of_nodes = reduced_model_part.NumberOfNodes();
+
+            // 2. create the result vector
+            Vector rResult;
+            rResult.resize((number_of_nodes)*3);
+
+            IndexType i = 0;
+            for (auto& r_node : reduced_model_part.Nodes()) {
+
+                const IndexType index = i * 3;
+                
+                rResult[index]     = r_node.GetDof(DISPLACEMENT_X).EquationId();
+                rResult[index + 1] = r_node.GetDof(DISPLACEMENT_Y).EquationId();
+                rResult[index + 2] = r_node.GetDof(DISPLACEMENT_Z).EquationId();
+
+                i++;
+            }
+
+            // 3. assigned the value of the reduced stifness and stabilization matrices based on result vector
+            reduced_stabilization_matrix = ZeroMatrix((number_of_nodes)*3,(number_of_nodes)*3);
+
+            for (IndexType i = 0; i < (number_of_nodes)*3; i++) 
+            {
+                for (IndexType j = 0; j <= i; j++) 
+                {
+                    reduced_stabilization_matrix(i,j) = rStabilizationMatrix(rResult(i),rResult(j));
+                    if (i != j)
+                        reduced_stabilization_matrix(j,i) = rStabilizationMatrix(rResult(i),rResult(j));
```

Comment:
```suggestion
                    if (i != j) {
                        reduced_stabilization_matrix(j,i) = rStabilizationMatrix(rResult(i),rResult(j));
                    }
```

try to always keep the brackets. It is also somewhere KRATOS-style guide.